src/python/gem5/resources/resource.py

```diff
@@ -963,13 +966,14 @@ def obtain_resource(
                     ``resource_directory`` parameter.
     :param quiet: If ``True``, suppress output. ``False`` by default.
     """
-
+    proxy_context = get_proxy_context()
```

Comment:
Why do we need to get the proxy_context here? Can't we get this in the `get_resource_json` function? Can't we just call this `get_proxy_context` function when it's needed? This approach requires changing quite a lot of function signatures and passing data around.

Reply:
I did this because we only want to change the context once and then pas the same context around.

Reply:
This causes an import error due to circular imports, as the downloader.py needs to access a function from client.py and client.py imports the AltasClient which in turn import get_proxy_context() from downloader.py.